### ISSUE TITLE
AstroLSP not autostarting language server defined by `opts.servers`

### ISSUE THREAD

**Issue by cactusbento** [AstroNvim/AstroNvim#2522]
### Checklist

- [X] I have searched through the AstroNvim docs
- [X] I have searched through the existing issues of AstroNvim
- [X] I have searched the existing issues of plugins related to this issue

### Neovim version (nvim -v)

0.9.5

### Operating system/version

Arch Linux KDE / pacman -Syu as of post date

### Terminal/GUI

konsole

### Describe the bug

Language servers defined following https://docs.astronvim.com/recipes/advanced_lsp/#lsp-setup-without-installer will not autostart even if `:LspInfo` states that `autostart: true`.

Servers installed by `:LspInstall` or `:MasonInstall` will autostart.

### Steps to Reproduce

1. Add language server to `$PATH` (zls)
2. Define installed server:
```lua
-- lua/plugins/astrolsp.lua

return {
    "AstroNvim/astrolsp",
    opts = {
        servers = { 
            "zls", 
        },    
    },
}
```
3. 
```bash
nvim main.zig # on existing file, or
nvim niam.zig # on non-existing file
```

### Expected behavior

Zig Language Server autostarts from 
```bash
nvim main.zig # on existing file, or
nvim niam.zig # on non-existing file
```

### Additional Context

`:LspInfo` shows that the correct executable path is detected.

This still persists when used with an `opts` function.

### Repro

```Lua
-- save as repro.lua
-- run with nvim -u repro.lua
-- DO NOT change the paths
local root = vim.fn.fnamemodify("./.repro", ":p")

-- set stdpaths to use .repro
for _, name in ipairs({ "config", "data", "state", "runtime", "cache" }) do
  vim.env[("XDG_%s_HOME"):format(name:upper())] = root .. "/" .. name
end

-- bootstrap lazy
local lazypath = root .. "/plugins/lazy.nvim"
if not vim.loop.fs_stat(lazypath) then
  -- stylua: ignore
  vim.fn.system({ "git", "clone", "--filter=blob:none", "https://github.com/folke/lazy.nvim.git", "--branch=stable", lazypath })
end
vim.opt.rtp:prepend(vim.env.LAZY or lazypath)

-- install plugins
local plugins = {
  { "AstroNvim/AstroNvim", tag = "v4.1.6", import = "astronvim.plugins" },
  -- add any other plugins/customizations here
  {
  "AstroNvim/astrolsp",
    opts = {
      servers = {
        "zls",
      },
    },
  },
}
require("lazy").setup(plugins, {
  root = root .. "/plugins",
})

-- add anything else here (autocommands, vim.filetype, etc.)
```

**Comment by mehalter** [comment]
Could you please test the latest push to `main` and see if this is working now? Just remove the `tag`/`version` from your "AstroNvim/AstroNvim" definition and do `:Lazy update`

**Comment by mehalter** [comment]
This should be fixed in AstroNvim v4.1.7 (https://github.com/AstroNvim/AstroNvim/commit/eceb0a803bbaf146b0a7a33885b4d7cf410a345d) which is available now!

**Comment by cactusbento** [comment]
Yep, it's fixed now. Thanks!
